File: 2006Scape Server/plugins/plugin/buttons/gameframe/LogoutButton.java
package plugin.buttons.gameframe;

import com.rs2.event.SubscribesTo;
import com.rs2.event.impl.ButtonActionEvent;
import com.rs2.game.players.Player;
import plugin.buttons.ButtonClick;

@SubscribesTo(ButtonActionEvent.class)
public final class LogoutButton extends ButtonClick {

    @Override
    protected void execute(Player player, ButtonActionEvent event) {
        switch (event.getButton()) {

            case 9154:
                player.logout();
                break;

        }
    }

    @Override
    public boolean test(ButtonActionEvent event) {
        return event.getButton() == 9154;
    }

}

File: 2006Scape Server/plugins/plugin/buttons/gameframe/LogoutButton.kt
package plugin.buttons.gameframe

import com.rs2.event.SubscribesTo
import com.rs2.event.impl.ButtonActionEvent

import com.rs2.game.players.Player
import plugin.buttons.ButtonClick

@SubscribesTo(ButtonActionEvent::class)
class LogoutButton : ButtonClick() {

	override fun execute(player: Player, event: ButtonActionEvent) {
		player.logout()
	}

	override fun test(event: ButtonActionEvent): Boolean {
		return event.button == 9154
	}

}